docs/integrations/delta-lake-dagster.md

```diff
@@ -0,0 +1,278 @@
+
+The Delta Lake I/O Manager will handle reading and writing the data from your Delta Lake.
+
+```
+import pyarrow as pa
+from dagster import asset
+
+# ... raw_dataset asset is defined here ...
+
+@asset
+def clean_dataset(raw_dataset: pa.Table) -> pa.Table:
+   return raw_dataset.drop_null()
+```
+
+## Reading Existing Delta Lake Tables into Dagster
+You can make existing Delta Lake tables (that were not created in Dagster) available to your Dagster assets. Use the `SourceAsset` object and pass the table name as the key argument: 
+
+```
+from dagster import SourceAsset
+
+iris_harvest_data = SourceAsset(key="more_animal_data")
+```
+
+This will load a table `more_animal_data` located at `<root_uri>/<schema>` as configured in the Definitions object above (see [Delta Lake I/O Manager](#the-delta-lake-io-manager) section).
+
+## Column Pruning
+You can often improve the efficiency of your computations by only loading specific columns of your Delta table. This is called column pruning. 
+
+With the Delta Lake I/O manager, you can select specific columns to load defining the `columns` in the `metadata` parameter of the `AssetIn` that loads the upstream Asset:
+
+```
+import pyarrow as pa
+from dagster import AssetIn, asset
+
+# this example uses the clean_dataset Asset defined earlier
+
+@asset(
+       ins={
+           "mammal_bool": AssetIn(
+               key="clean_dataset",
+               metadata={"columns": ["is_mammal", "animals"]},
+           )
+       }
+)
+def mammal_data(mammal_bool: pa.Table) -> pa.Table:
+   mammals = mammal_bool["is_mammal"].cast("bool")
+   animals = mammal_bool["animals"]
+   data = {"mammal_bool": mammals, "animals": animals}
+   return pa.Table.from_pydict(data)
+```
+
+Here, we select only the `sepal_length_cm` and `sepal_width_cm` columns from the `iris_dataset` table and load them into an `AssetIn` object called `iris_sepal`. This AssetIn object is used to create a new Asset `sepal_data`, containing only the selected columns.
+
+## Working with Partitioned Assets
+Partitioning is an important feature of Delta Lake that can make your computations more efficient. The Delta Lake I/O manager helps you read and write partitioned data easily. You can work with static partitions, time-based partitions, multi-partitions, and dynamic partitions.
+
+For example, you can partition the Iris dataset on the `species` column as follows:
+
+```
+import pyarrow as pa
+
+from dagster import StaticPartitionsDefinition, asset
+
+@asset(
+  partitions_def=StaticPartitionsDefinition(
+      ["Human", "Horse",]
+  ),
+  metadata={"partition_expr": "n_legs"},
+)
+def dataset_partitioned(
+   context,
+   clean_dataset: pa.Table,
+   ) -> pa.Table:
+   animals = context.asset_partition_key_for_output()
+   table = clean_dataset
+
+   return table.filter(pc.field("animals") == animals)
```

Comment:
Having trouble getting the PyArrow code in the partitioning section to work. I must be overlooking something. Would very much appreciate an extra set of eyes to help me find my blind spot :slightly_smiling_face: 

The pandas version listed [here](https://docs.dagster.io/integrations/deltalake/reference#storing-static-partitioned-assets) works fine.